python/pacemaker/_cts/audits.py

```diff
@@ -291,13 +297,28 @@ def __call__(self):
         self._cm.ns.wait_for_all_nodes(self._cm.env["nodes"])
 
         for node in self._cm.env["nodes"]:
-            found = self._find_core_on_fs(node, "/var/lib/pacemaker/cores/*", "Pacemaker")
-            if found:
-                passed = False
+            found = False
 
-            found = self._find_core_on_fs(node, "/var/lib/corosync", "Corosync")
-            if found:
-                passed = False
+            # If systemd is present, first see if coredumpctl logged any core dumps.
+            if self._cm.env["have_systemd"]:
+                found = self._find_core_with_coredumpctl(node)
+                if found:
+                    passed = False
+
+            # If we didn't find any core dumps, it's for one of three reasons:
+            # (1) Nothing crashed
+            # (2) systemd is not present
+            # (3) systemd is present but coredumpctl is not enabled
+            #
+            # To handle the last two cases, check the other filesystem locations.
+            if not found:
```

Comment:
I think this is fine.

It duplicates work in the common case, which I don't like. If systemd is present, coredumpctl is enabled, and there were no crashes found via coredumpctl, we're still checking the filesystem locations every time. But we make a lot of rsh connections no matter what.